home/tests/test_serializers.py

```diff
@@ -0,0 +1,285 @@
+import pytest
+
+from home.models import ContentPage, HomePage
+from home.serializers import (
+    format_whatsapp_message,
+    has_next_message,
+    has_previous_message,
+)
+
+from .page_builder import (
+    MBlk,
+    MBody,
+    PageBuilder,
+    SBlk,
+    SBody,
+    UBlk,
+    UBody,
+    VarMsg,
+    VBlk,
+    VBody,
+    WABlk,
+    WABody,
+)
+
+
+def create_content_page(
+    wa_body_count: int = 1,
+    wa_gender_var: list[str] | None = None,
+) -> ContentPage:
+    """
+    Helper function to create pages needed for each test.
+
+    Parameters
+    ----------
+    wa_body_count : int
+        How many WhatsApp message bodies to create on the content page.
+    wa_gender_var: list[str]
+        Variation restriction
+    """
+    title = "default page"
+    home_page = HomePage.objects.first()
+    main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
+    parent = main_menu
+
+    if wa_gender_var is None:
+        wa_gender_var = []
+
+    bodies = [
+        MBody(title, [MBlk("Messenger 1"), MBlk("Messenger 2")]),
+        SBody(title, [SBlk("SMS 1"), SBlk("SMS 2")]),
+        UBody(title, [UBlk("USSD 1"), UBlk("USSD 2")]),
+        VBody(title, [VBlk("Viber 1"), VBlk("Viber 2")]),
+    ]
+
+    for i in range(wa_body_count):
+        wa_body = f"WhatsApp {i+1}"
+
+        variation_messages = [
+            VarMsg(f"WhatsApp Varied {var+1}", gender=wa_gender_var[var])
+            for var in range(len(wa_gender_var))
```

Comment:
`[VarMsg(f"WhatsApp Varied {i}", gender=gender) for i, gender in enumerate(wa_gender_var, 1)]` perhaps?